test/e2e/mcd_test.go

```diff
@@ -309,12 +309,15 @@ func TestReconcileAfterBadMC(t *testing.T) {
 		}
 		for _, node := range nodes {
 			if node.Annotations[constants.DesiredMachineConfigAnnotationKey] == newMCName && node.Annotations[constants.MachineConfigDaemonStateAnnotationKey] != constants.MachineConfigDaemonStateDone {
-				return true, nil
+				// just check that we have the annotation here, w/o strings checking anything that can flip fast causing flakes
+				if node.Annotations[constants.MachineConfigDaemonReasonAnnotationKey] != "" {
```

Comment:
i stared at this and can't figure out what this check is doing here? help.

Reply:
makes sure that a Reason for failure is set if this is the node that is in unreconcilable state (just an added check to verify the code in this PR)